### ISSUE TITLE
Provide information for new developers/testers

### ISSUE THREAD

**Issue by stohrendorf** [opentomb/OpenTomb#383]
Should contain information about how to compile from scratch on different platforms, and also a few style guidelines about code formatting, variable/type naming, etc.

**Comment by Gh0stBlade** [comment]
Coding style guidelines were posted by Lwmte here: https://github.com/opentomb/OpenTomb/issues/180

**Comment by stohrendorf** [comment]
Oh, I totally forgot about that. But it should rather be on its own document (e.g. a wiki page) than a github issue.

**Comment by Legend286** [comment]
Thanks for the response @Gh0stBlade :) I'm a bit occupied myself atm but it would be rad to collaborate with someone on the renderer stuff because I'm not so familiar with opengl-centric c++ (obviously shaders are a universal thing as a TA) but it should be interesting to see exactly how much we can get out of the old art with some deferred lighting, shadowmapping, etc etc... and maybe it'll be cool for people who still use TRLE (my friend is a big TRLE map maker so part of the reason I want to undertake this kind of project is to have a bit of fun but also because it'd be cool to see what he does with it)

**Comment by CDrosos** [comment]
I cant understand how to run this project on windows.
Tr1 does not have a data folder, i have just copy paste all the find on the tr1/data folder
music files cannot be downloaded, it says forbiten,
i have extract the loading screens, i have take the data folders from tomb raider 2 and 3 and when i run opentomb i get nothing, we need better installation info

**Comment by Gh0stBlade** [comment]
Which version are of TR1 do you have? GOG?

**Comment by CDrosos** [comment]
yes all are GOG

**Comment by Gh0stBlade** [comment]
https://github.com/opentomb/OpenTomb/issues/434

GOG has TR1 level files stored in another file which must be unpacked, the above issue linked details how to do this.

**Comment by CDrosos** [comment]
I have extract the .GOG file.
i have put the contents of the DATA folder on open\data\tr1\data but i still get a blank screen

**Comment by Gh0stBlade** [comment]
Press ` to open the console window.

Type setgamef(1.0, 1) (TR1, Level1). For TR2 you'd do:
setgamef(2.0, 1)

**Comment by CDrosos** [comment]
where are my guns? 🗡 :D

**Comment by Gh0stBlade** [comment]
Config.lua lists all key bindings "bind(act.drawweapon, KEY_F);"

Pressing F should draw weapons.

**Comment by CDrosos** [comment]
im a c#,php developer, can i help somehow on this project? do you have a page for this cause?

**Comment by Gh0stBlade** [comment]
OpenTomb is written in C/C++ and also uses LUA scripts which can customise most of the engine/game logic.

There is no page specifically requesting developers to help. You can just fork the project, make changes and create pull requests of any bug fixes/feature enhancements you wish to add. Note: That does not mean the code will be merged, it is at discretion of the lead programmer.

**Comment by rsantellan** [comment]
@Gh0stBlade I looking to play TR1 and learn in the process. One thing that I can't find is the status of the project:

- what is developed? 
- what is not? 
- What is missing?
- Improved stuff.

I take the ideas from this project: https://github.com/SFTtech/openage/blob/master/doc/status.md

**Comment by vvs-** [comment]
There is no official TODO document, except some development notes here: https://github.com/opentomb/OpenTomb/blob/master/TODO.md You can use these GitHub issues as a rough guidelines: https://github.com/opentomb/OpenTomb/issues?q=is%3Aopen+is%3Aissue+label%3ATodo

**Comment by rsantellan** [comment]
@vvs- I downloaded the code and Code Blocks IDE. 
Just a newbie question, after I make a change do I have to run each time the cmake . and make? Or just the make is enough? Is there a way to trigger it from the IDE? 
Regards.

**Comment by vvs-** [comment]
You don't need to run CMake each time, make will do it for you automatically. I'm not using any IDE, but as far as I know you can't use CMake project from it. You should build a Code::Blocks project using CMake first and that's the only project format that's possible for IDE to use.

**Comment by stohrendorf** [comment]
No, CMake can also generate project files for VS. It adds an artificial project "ZERO_CHECK", which is a dependency of all other projects and only calls CMake to update the other projects if necessary. QtCreator works better with CMake, but uses Code::Blocks project files as the backend. KDevelop has a nice CMake integration. Clion isn't that great when it comes to customizing the CMake call. You can also try to use Eclipse or Netbeans with C++ support.

**Comment by vvs-** [comment]
I didn't mean that CMake can't generate other types of projects. I meant that Code::Blocks can only use its own projects and isn't CMake aware.
